templates/common/gcp/files/usr-local-bin-update-dns-server.yaml

```diff
@@ -0,0 +1,22 @@
+mode: 0755
+path: "/usr/local/bin/update-dns-server"
+contents:
+  inline: |
+    #!/bin/bash
+    # This is test code
+    
+    echo "Executing update-dns-server"
+
+    mkdir -p /etc/NetworkManager/conf.d
+
+    echo "#Added by OpenShift" > /etc/NetworkManager/conf.d/dns-servers.conf
+    echo "[global-dns-domain-*]" >> /etc/NetworkManager/conf.d/dns-servers.conf
+
+    # 169.254.169.254 is the IP address of the GCP metadata server
+    echo "servers=$(ip route get 8.8.8.8 | head -1 | cut -d' ' -f7),169.254.169.254" >> /etc/NetworkManager/conf.d/dns-servers.conf
```

Comment:
I would use parse the json here rather than using head and cut since that's an option for us: 

```
[core@cosa-devsh ~]$ ip -j route get 8.8.8.8
[{"dst":"8.8.8.8","gateway":"10.0.2.2","dev":"ens5","prefsrc":"10.0.2.15","flags":[],"uid":1000,"cache":[]}]
```

Reply:
Done.